zebROS_ws/src/behaviors/src/2023_placing_server.cpp

```diff
@@ -0,0 +1,237 @@
+				break;
+			}
+
+			if (ac.getState().isDone()) {
+				break;
+			}
+			r.sleep();
+		}
+
+		return ac.getState().isDone();
+	}
+
+	void gamePieceCallback(const std_msgs::UInt8 msg) {
+		latest_game_piece_ = msg.data;
+		latest_game_piece_time_ = ros::Time::now();
+	}
+
+	void gamePiecePositionCallback(const std_msgs::Float64 msg) {
+		latest_game_piece_position_ = msg.data;
+		latest_game_piece_position_time_ = ros::Time::now();
+	}
+
+	void executeCB(const behavior_actions::Placing2023GoalConstPtr &goal)
+	{
+		if (!ac_fourber_.waitForServer(ros::Duration(server_timeout_))) {
+			ROS_ERROR_STREAM("2023_placing_server : timed out connecting to fourber server, aborting");
+			result_.timed_out = true;
+			as_.setAborted(result_);
+			return;
+		}
+
+		if (!ac_elevater_.waitForServer(ros::Duration(server_timeout_))) {
+			ROS_ERROR_STREAM("2023_placing_server : timed out connecting to elevater server, aborting");
+			result_.timed_out = true;
+			as_.setAborted(result_);
+			return;
+		}
+
+		if (!ac_intake_.waitForServer(ros::Duration(server_timeout_))) {
+			ROS_ERROR_STREAM("2023_placing_server : timed out connecting to intake server, aborting");
+			result_.timed_out = true;
+			as_.setAborted(result_);
+			return;
+		}
+
+		if (ros::Time::now() - latest_game_piece_time_ > ros::Duration(game_piece_timeout_)) {
+			ROS_ERROR_STREAM("2023_placing_server : game piece data too old, aborting");
+			result_.timed_out = true;
+			as_.setAborted(result_);
+			return;
+		}
+
+		if (ros::Time::now() - latest_game_piece_position_time_ > ros::Duration(game_piece_timeout_)) {
+			ROS_ERROR_STREAM("2023_placing_server : game piece position data too old, aborting");
+			result_.timed_out = true;
+			as_.setAborted(result_);
+			return;
+		}
+
+		if (goal->align_intake) {
+			path_follower_msgs::holdPositionGoal hold_position_goal_;
+			hold_position_goal_.pose.orientation.x = 0.0;
+			hold_position_goal_.pose.orientation.y = 0.0;
+			hold_position_goal_.pose.orientation.z = 0.0;
+			hold_position_goal_.pose.orientation.w = 1.0;
+
+			hold_position_goal_.pose.position.x = 0.0;
+			hold_position_goal_.pose.position.y = -latest_game_piece_position_;
+			hold_position_goal_.pose.position.z = 0.0;
+
+			hold_position_goal_.isAbsoluteCoord = false;
+
+			ROS_INFO_STREAM("2023_placing_server : holding position, y = " << hold_position_goal_.pose.position.y);
+
+			ac_hold_position_.sendGoal(hold_position_goal_);
+
+			if (!waitForResultAndCheckForPreempt(ros::Duration(-1), ac_hold_position_, as_)) {
+				ROS_INFO_STREAM("2023_placing_server : hold position server timed out, aborting");
+				result_.timed_out = true;
+				as_.setAborted(result_);
+				ac_hold_position_.cancelGoalsAtAndBeforeTime(ros::Time::now());
+				return;
+			}
+		}
+
+		behavior_actions::Fourber2023Goal fourber_goal;
+		fourber_goal.piece = latest_game_piece_;
+		fourber_goal.mode = goal->node + 1; // please don't change the actionlib files
```

Comment:
This is going to break at some point.  Probably best to do an explicit conversion. Shouldn't be too bad, just a switch statement checking input and setting output

Reply:
Make it a function so it can be reused below.
And add in a default for cases which don't match so this can be easily debugged.